frontend/src/pages/Archive.vue

```diff
@@ -293,297 +293,263 @@
+  deletePopup: false
+})
+const uploadedFiles = ref(null)
+const editorText = ref('')
+const oldName = ref('')
+const file = ref({
+  name: '',
+  progress: 0
+})
+const itemToDelete = ref(null)
 
-  watch: {
-    async connected (newStatus, oldStatus) {
-      if (newStatus) {
-        await this.start()
-      }
-    }
-  },
+watch(() => props.connected, (newStatus) => {
+  if (newStatus) {
+    start()
+  }
+})
 
-  methods: {
-    async startRpc () {
-      this.flags.rpcToggling = true
-      await this.flipper.startRPCSession()
-        .catch(error => {
-          console.error(error)
-          this.$emit('log', {
-            level: 'error',
-            message: `${this.componentName}: Error while starting RPC: ${error.toString()}`
-          })
-        })
-      this.flags.rpcActive = true
-      this.$emit('setRpcStatus', true)
-      this.flags.rpcToggling = false
-      this.$emit('log', {
+const startRpc = async () => {
+  flags.value.rpcToggling = true
+  await props.flipper.startRPCSession()
+    .then(() => {
+      flags.value.rpcActive = true
+      emit('setRpcStatus', true)
+      flags.value.rpcToggling = false
+      emit('log', {
         level: 'info',
-        message: `${this.componentName}: RPC started`
+        message: `${componentName}: RPC started`
       })
-    },
-
-    async stopRpc () {
-      this.flags.rpcToggling = true
-      await this.flipper.setReadingMode('text', 'promptBreak')
-      this.flags.rpcActive = false
-      this.$emit('setRpcStatus', false)
-      this.flags.rpcToggling = false
-      this.$emit('log', {
-        level: 'info',
-        message: `${this.componentName}: RPC stopped`
+    })
+    .catch(error => {
+      console.error(error)
+      emit('log', {
+        level: 'error',
+        message: `${componentName}: Error while starting RPC: ${error.toString()}`
       })
-    },
-
-    async list () {
-      const list = await this.flipper.RPC('storageList', { path: this.path })
-        .catch(error => this.rpcErrorHandler(error, 'storageList'))
-        .finally(() => {
-          this.$emit('log', {
-            level: 'debug',
-            message: `${this.componentName}: storageList: ${this.path}`
-          })
-        })
-      if (list.length === 0) {
-        this.dir = []
-        return
-      }
-
-      if (this.path === '/') {
-        this.dir = list.filter(e => e.name !== 'any')
-      } else {
-        this.dir = list
-      }
-    },
-
-    async read (path, preventDownload) {
-      this.flags.blockingOperationPopup = true
-      this.file.name = path.slice(path.lastIndexOf('/') + 1)
-      const file = this.dir.find(e => e.name === this.file.name && !e.type)
-      const total = file.size
-      const unbind = this.flipper.emitter.on('storageReadRequest/progress', chunks => {
-        this.file.progress = Math.min(chunks * 512, total) / total
+    })
+}
+
+const list = async () => {
+  const list = await props.flipper.RPC('storageList', { path: path.value })
```

Comment:
Since you have `then` here now, then you need to assign the `list` variable inside `then`. That is

```
let list = []
await props.flipper.RPC('storageList', { path: path.value })
  .then((res) => {
    list = res
    ...
```